server/src/test/java/org/apache/iotdb/db/tools/TsFileSelfCheckToolTest.java

```diff
@@ -0,0 +1,226 @@
+        measurementSchemas.add(measurementSchema);
+        schema.registerTimeseries(
+            new Path(device),
+            new MeasurementSchema(sensorPrefix + (i + 1), TSDataType.INT64, TSEncoding.TS_2DIFF));
+      }
+
+      // add measurements into TSFileWriter
+      try (TsFileWriter tsFileWriter = new TsFileWriter(f, schema)) {
+
+        // construct the tablet
+        Tablet tablet = new Tablet(device, measurementSchemas);
+
+        long[] timestamps = tablet.timestamps;
+        Object[] values = tablet.values;
+
+        long timestamp = 1;
+        long value = 1000000L;
+
+        for (int r = 0; r < rowNum; r++, value++) {
+          int row = tablet.rowSize++;
+          timestamps[row] = timestamp++;
+          for (int i = 0; i < sensorNum; i++) {
+            long[] sensor = (long[]) values[i];
+            sensor[row] = value;
+          }
+          // write Tablet to TsFile
+          if (tablet.rowSize == tablet.getMaxRowNumber()) {
+            tsFileWriter.write(tablet);
+            tablet.reset();
+          }
+        }
+        // write Tablet to TsFile
+        if (tablet.rowSize != 0) {
+          tsFileWriter.write(tablet);
+          tablet.reset();
+        }
+      }
+    } catch (Exception e) {
+      throw new Exception("meet error in TsFileWrite with tablet", e);
+    }
+  }
+
+  @Test
+  public void tsFileSelfCheckToolCompleteTest() {
+    TsFileSelfCheckTool tool = new TsFileSelfCheckTool();
+    try {
+      tool.check(path, false);
+    } catch (IOException
+        | TsFileStatisticsMistakesException
+        | TsFileTimeseriesMetadataException e) {
+      fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void tsFileSelfCheckToolWithStatisticsModifiedTest()
+      throws IOException, TsFileTimeseriesMetadataException {
+    Map<Long, Pair<Path, TimeseriesMetadata>> timeseriesMetadataMap =
+        new TsFileSelfCheckTool().getTimeseriesMetadataMapWithPath(path);
+    for (Map.Entry<Long, Pair<Path, TimeseriesMetadata>> entry : timeseriesMetadataMap.entrySet()) {
+      TimeseriesMetadata timeseriesMetadata = entry.getValue().right;
+      Long pos = entry.getKey();
+      LongStatistics statistics = (LongStatistics) timeseriesMetadata.getStatistics();
+      statistics.initializeStats(666, 1999999, 1000000, 1999999, 0);
+
+      RandomAccessFile raf = new RandomAccessFile(path, "rw");
+      ByteArrayOutputStream bo = new ByteArrayOutputStream();
+      int serialLength = ReadWriteIOUtils.write(timeseriesMetadata.getTimeSeriesMetadataType(), bo);
+      serialLength += ReadWriteIOUtils.writeVar(timeseriesMetadata.getMeasurementId(), bo);
+      serialLength += ReadWriteIOUtils.write(timeseriesMetadata.getTSDataType(), bo);
+      serialLength +=
+          ReadWriteForEncodingUtils.writeUnsignedVarInt(
+              timeseriesMetadata.getDataSizeOfChunkMetaDataList(), bo);
+      serialLength += statistics.serialize(bo);
+      System.out.println("serialLength: " + serialLength);
```

Comment:
I think use` logger` instead of `System.out.println` is better.

Reply:
Hi, why logger.info can not output to terminal? logger.error can output to terminal. Is it idea setting problem?